src/WasmSdk/Tasks/BootJsonData.cs

```diff
@@ -0,0 +1,145 @@
+
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using ResourceHashesByNameDictionary = System.Collections.Generic.Dictionary<string, string>;
+
+namespace Microsoft.NET.Sdk.WebAssembly
+{
+#pragma warning disable IDE1006 // Naming Styles
+    /// <summary>
+    /// Defines the structure of a Blazor boot JSON file
+    /// </summary>
+    public class BootJsonData
+    {
+        /// <summary>
+        /// Gets the name of the assembly with the application entry point
+        /// </summary>
+        public string entryAssembly { get; set; }
+
+        /// <summary>
+        /// Gets the set of resources needed to boot the application. This includes the transitive
+        /// closure of .NET assemblies (including the entrypoint assembly), the dotnet.wasm file,
+        /// and any PDBs to be loaded.
+        ///
+        /// Within <see cref="ResourceHashesByNameDictionary"/>, dictionary keys are resource names,
+        /// and values are SHA-256 hashes formatted in prefixed base-64 style (e.g., 'sha256-abcdefg...')
+        /// as used for subresource integrity checking.
+        /// </summary>
+        public ResourcesData resources { get; set; } = new ResourcesData();
+
+        /// <summary>
+        /// Gets a value that determines whether to enable caching of the <see cref="resources"/>
+        /// inside a CacheStorage instance within the browser.
+        /// </summary>
+        public bool cacheBootResources { get; set; }
+
+        /// <summary>
+        /// Gets a value that determines if this is a debug build.
+        /// </summary>
+        public bool debugBuild { get; set; }
+
+        /// <summary>
+        /// Gets a value that determines if the linker is enabled.
+        /// </summary>
+        public bool linkerEnabled { get; set; }
+
+        /// <summary>
+        /// Config files for the application
+        /// </summary>
+        public List<string> config { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ICUDataMode"/> that determines how icu files are loaded.
+        /// </summary>
+        public ICUDataMode icuDataMode { get; set; }
+
+        /// <summary>
+        /// Gets or sets configuration extensions.
+        /// </summary>
+        public Dictionary<string, Dictionary<string, object>> extensions { get; set; }
+    }
+
+    public class ResourcesData
+    {
+        /// <summary>
+        /// .NET Wasm runtime resources (dotnet.wasm, dotnet.js) etc.
+        /// </summary>
+        public ResourceHashesByNameDictionary runtime { get; set; } = new ResourceHashesByNameDictionary();
+
+        /// <summary>
+        /// "assembly" (.dll) resources
+        /// </summary>
+        public ResourceHashesByNameDictionary assembly { get; set; } = new ResourceHashesByNameDictionary();
+
+        /// <summary>
+        /// "debug" (.pdb) resources
+        /// </summary>
+        [DataMember(EmitDefaultValue = false)]
```

Comment:
FYI, this can be updated to use `System.Text.Json`. It used datacontractserializer because it predates it, but I did support to add System.Text.Json support to the SDK tasks, so you should be able to use it. (It might be that I only did it in the RazorSDK, but it should be "copiable")

Reply:
Not asking you to change it, just letting you know

Reply:
I would love to do it. I think I struggled on the package version when I tried it last week. So you are saying that I can use version 6.0 like in here https://github.com/dotnet/razor/blob/33e06ed0fd64c8b67b80d464370ee0380f9ab32c/eng/Versions.props#L179, right?

Reply:
https://github.com/dotnet/sdk/commit/9ed247b28cba4501185dad4e546dc9c51e1c5711#diff-8bd75765f34d87f1dcad4a428355ceb7e78e7be3febb15e66bbae735c60ae374R42